facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java

```diff
@@ -357,4 +397,19 @@ public TimeoutConfiguration getTimeoutConfiguration() {
     public void setTimeoutConfiguration(TimeoutConfiguration timeoutConfiguration) {
         this.timeoutConfiguration = timeoutConfiguration;
     }
+
+    /**
+     * Shutdown thread pool executor for all facades.
+     *
+     */
+    public static void shutdown() {
```

Comment:
We don't restart in all envs after update, but your point about this not
knowing its lifecycle covers it.  The EJB would need updated to hook into
this.  Thanks, knew I was missing something.

On Wed, May 18, 2016 at 3:02 PM, Marek <notifications@github.com> wrote:

> In
> facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java
> <https://github.com/lightblue-platform/lightblue-migrator/pull/350#discussion_r63762293>
> :
>
> > @@ -357,4 +397,19 @@ public TimeoutConfiguration getTimeoutConfiguration() {
> >      public void setTimeoutConfiguration(TimeoutConfiguration timeoutConfiguration) {
> >          this.timeoutConfiguration = timeoutConfiguration;
> >      }
> > +
> > +    /**
> > +     * Shutdown thread pool executor for all facades.
> > +     *
> > +     */
> > +    public static void shutdown() {
>
> I'm not sure if we need to worry about this, since we restart app server
> after every change. It's not that much of a problem if there is a leak
> during hot deployment and we know how to fix it (a singleton to shutdown
> facade on pre-destroy) if we decide to do so.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-migrator/pull/350/files/17b676bee35728c0487e52415290b9e02d5b9cc0#r63762293>
>